File: Benchmark-Application/benchmarks/AccesareMemorieDinamica.py
import time
import numpy as np

ARRAY_SIZE = 10000
n = 100


def exec_program():
    # Record the start time
    start_time = time.time()

    # Run the program n times
    for k in range(n):
        # Perform dynamic memory allocation
        dynamic_array = np.arange(1, ARRAY_SIZE + 1, dtype=int)

        # Release the dynamically allocated memory (not required in Python, as it is managed by the garbage collector)
        # No equivalent of "delete[]" in Python

    # Record the end time
    end_time = time.time()

    # Calculate the elapsed time in seconds
    elapsed_time = end_time - start_time

    print(f"{elapsed_time:.2e}")


if __name__ == "__main__":
    exec_program()

File: Benchmark-Application/benchmarks/AccesareMemorieDinamica.cpp
#include <iostream>
#include <stdio.h>
#include <ctime>
#include <chrono>

using namespace std;
using namespace chrono;

const int ARRAY_SIZE = 10000;
const int n = 100;
// java nanotime
// c++ high resolution clock
void exec()
{
    duration<double> totalTime;

    // Run the program n times
    for (long k = 0; k < n; k++)
    {
        // Perform dynamic memory allocation
        int* dynamicArray = new int[ARRAY_SIZE];

        // Record the start time
        high_resolution_clock::time_point start_time = high_resolution_clock::now();
        for (int i = 0; i < ARRAY_SIZE; i++)
        {
            dynamicArray[i] = i + 1;
        }

        high_resolution_clock::time_point end_time = high_resolution_clock::now();
        duration<double> elapsed_time = duration_cast<duration<double>>(end_time - start_time);

        // Release the dynamically allocated memory
        delete[] dynamicArray;
        //cout << scientific << elapsed_time.count() << endl;
        totalTime += elapsed_time;
    }

    cout << scientific << totalTime.count();
}

int main()
{
    exec();


    return 0;
}
